### PR TITLE
[UXIT-1740] Event Page · Event Section

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description
This PR introduces the new Event Section.

- **Type:** New feature / Refactor

Note:
 - The feature doesn't include event images like the design does - [Slack Thread](https://filecoinfoundation.slack.com/archives/C06MTEQ3SP6/p1732167788920009)
- The feature uses the same date formatting as on `/events` 

## 🛠️ Key Changes
- Created `EventBaseFrontMatterSchema` for validating event metadata
- Added `EventSectionSchema` to handle event section configuration
- Made date utility functions generic with `DateFields` type
- Updated event section config to support new schema
- Added Bangkok 2024 event data using new schema
- Adapted Event Section component to work with new data structure

## 📌 To-Do Before Merging
- [ ] Verify all event data validates against new schemas
- [ ] Check event sorting and filtering still works correctly
- [ ] Ensure Bangkok 2024 event displays properly

## 🧪 How to Test
- **Setup:** No special setup required

- **Steps to Test:**
  1. Check Event Section page displays correctly
  2. Verify Bangkok 2024 event appears in the right section
  3. Confirm event sorting works (newest/oldest)
  4. Validate all event metadata displays correctly

- **Expected Results:** 
  - Event section should show properly formatted events
  - Events should be correctly sorted and filtered
  - All event metadata should be type-safe and validated
  - Date utilities should work with any object containing date fields

CleanShot 2024-11-21 at 12 54 10@2x

### CODE REVIEW

File: public/admin/config.yml [hunk 1]
```diff
@@ -737,11 +737,48 @@ collections:
...
-            label: "Luma Embed URL"
-            widget: "string"
-            required: false
-            hint: Paste the embed URL for your Luma event here to display the events directly on the page. Include relevant tags when appropriate, e.g., https://lu.ma/embed/calendar/cal-nlDvL4B7Ko1swF0/events?lt=light&tag=FIL%20Bangkok%202024.
+          - name: "events"
+            label: "Events"
+            widget: "list"
+            required: true
+            fields:
+              - name: "title"
```
Comment (mirhamasala) [inline public/admin/config.yml:745]: I've tried to abstract these but it didn't work.

File: public/admin/config.yml [hunk 2]
```diff
@@ -737,11 +737,48 @@ collections:
...
+                  - name: "url"
+                    label: "URL"
+                    widget: "string"
+                    required: false
+                  - name: "text"
+                    label: "Text"
+                    widget: "string"
+                    required: false
+                    default: "View Event Details"
+                    hint: "Text for the external link CTA. Please use proper capitalization. Defaults to 'View Event Details.'"
```
Comment (barbaraperic) [inline public/admin/config.yml:781]: Curious why we would need this to be customisable? Do we want different CTAs on the cards? If yes, could we handle the capitalisation part on our side?

File: public/admin/config.yml [hunk 3]
```diff
@@ -737,11 +737,48 @@ collections:
...
+                required: false
+                hint: "Last day of event"
+                picker_utc: true
+              - name: "location"
+                label: "Location"
+                widget: "string"
+              - name: "description"
+                label: "Description"
+                widget: "string"
+                required: false
```
Comment (barbaraperic) [inline public/admin/config.yml:765]: Should description be below the title?

File: src/app/events/[slug]/components/EventSection.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { PageSection } from '@/components/PageSection'
+
+import type { Event } from '../../types/eventType'
+import { getMetaData } from '../../utils/getMetaData'
+import { sortEventsAsc } from '../../utils/sortEvents'
+
+type EventSectionProps = NonNullable<Event['eventSection']>
+
+export function EventSection({ title, events }: EventSectionProps) {
+  const sortedEvents = sortEventsAsc(events)
```
Comment (barbaraperic) [inline src/app/events/[slug]/components/EventSection.tsx:14]: What do you think of passing already sorted events to `EventSection`? Meaning, moving the sorting to the parent?

File: src/app/events/schemas/EventSectionSchema.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+export const EventSectionSchema = z
+  .object({
+    title: z.string().optional(),
+    events: z
+      .array(
+        EventBaseFrontMatterSchema.extend({
+          location: z.string(),
+        }),
+      )
+      .nonempty(),
```
Comment (barbaraperic) [inline src/app/events/schemas/EventSectionSchema.ts:14]: nice 👌

**Review by CharlyMartin — CHANGES_REQUESTED** [review]
I have to say, this new design looks much better than the Luma embed ✨

File: public/admin/config.yml (hunk 3 above)
Comment (mirhamasala) [reply public/admin/config.yml:765]: Sure.

File: public/admin/config.yml [hunk 6]
```diff
@@ -737,11 +737,48 @@ collections:
             widget: "string"
             required: false
             hint: This is the title for the Events section header. For example, "Main Stage Events." Defaults to 'Events.'
-          - name: "embed-link"
-            label: "Luma Embed URL"
-            widget: "string"
-            required: false
-            hint: Paste the embed URL for your Luma event here to display the events directly on the page. Include relevant tags when appropriate, e.g., https://lu.ma/embed/calendar/cal-nlDvL4B7Ko1swF0/events?lt=light&tag=FIL%20Bangkok%202024.
+          - name: "events"
+            label: "Events"
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply public/admin/config.yml:741]: Yes, I agree. I was struggling with this as well. Program is a good option. It's a little confusing with `Schedule` but it's better than `Event`. I'll change everything.

File: src/app/events/utils/sortEvents.ts [hunk 7]
```diff
@@ -7,38 +7,40 @@ import {
...
-export function getUpcomingEvents(events: Array<Event>) {
+type DateFields = Pick<Event, 'startDate' | 'endDate'>
+
+export function getUpcomingEvents<T extends DateFields>(events: Array<T>) {
   const yesterday = startOfYesterday()
 
   const upcomingEvents = events.filter((event) => {
     const eventDate = event.endDate || event.startDate
-    return isAfter(eventDate, yesterday)
+    return isAfter(new Date(eventDate), yesterday)
```
[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/events/utils/sortEvents.ts:19]: Yes. I didn't really look at this beyond making it generic so I could use it. I guess we never refactored this util after switching to Zod. I'll have a look.

[the remark this answers is not in the document]
Comment (mirhamasala) [reply src/app/events/utils/sortEvents.ts:19]: Done and refactored a bit further. I also added a note to our Follow Up Dev work ticket to look at the rest of the codebase.

File: src/app/events/[slug]/components/EventSection.tsx (hunk 4 above)
Comment (mirhamasala) [reply src/app/events/[slug]/components/EventSection.tsx]: I'm ending very much up in TS hell - I'm going to move this forward and maybe we can have a look together on Monday.

File: public/admin/config.yml (hunk 3 above)
Comment (mirhamasala) [reply public/admin/config.yml:764]: Done.

File: public/admin/config.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (mirhamasala) [reply public/admin/config.yml:740]: Done.

**Pull request merged**